src/openai/lib/azure.py

```diff
@@ -55,11 +55,21 @@ def _build_request(
     ) -> httpx.Request:
         if options.url in _deployments_endpoints and is_mapping(options.json_data):
             model = options.json_data.get("model")
-            if model is not None and not "/deployments" in str(self.base_url):
+            if model is not None and "/deployments" not in str(self.base_url.path):
                 options.url = f"/deployments/{model}{options.url}"
 
         return super()._build_request(options)
 
+    @override
+    def _prepare_url(self, url: str) -> httpx.URL:
+        if "/deployments" in str(self.base_url.path) and url not in _deployments_endpoints:
+            merge_url = httpx.URL(url)
+            if merge_url.is_relative_url:
+                merge_path = f"{self.base_url.path.rsplit('/deployments', maxsplit=1)[0]}/{merge_url.path.lstrip('/')}"
```

Comment:
Rather than adding it and then removing it, did we consider adding it only when needed?

Reply:
Should we split on/find `'/deployments/'`? I guess it is unlikely that one would name a deployment `"deployments"` (and we do a maxsplit=1 so it is only a problem if we have a deployments "to the left" of the intended split point).

Reply:
Sure, the base_url property [enforces a trailing slash](https://github.com/openai/openai-python/blob/main/src/openai/_base_client.py#L617) so it will always be there, regardless of whether the user added it.